presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ShardMetadataRecordCursor.java

```diff
@@ -340,13 +364,12 @@ else if (temporalType.equals(TIMESTAMP)) {
         if (schemaNameDomain != null || tableNameDomain != null) {
             sql.append("WHERE ");
             List<String> predicates = new ArrayList<>();
-            if (tableNameDomain != null && tableNameDomain.isSingleValue()) {
-                predicates.add("table_name = ?");
-                values.add(getStringValue(tableNameDomain.getSingleValue()));
+            if (tableNameDomain != null ) {
+                buildPredicatesAndValues(tableNameDomain, values, predicates, "table_name");
             }
-            if (schemaNameDomain != null && schemaNameDomain.isSingleValue()) {
-                predicates.add("schema_name = ?");
-                values.add(getStringValue(schemaNameDomain.getSingleValue()));
+
+            if (schemaNameDomain != null) {
```

Comment:
Let's swap the order so that schema comes before table, as that is the natural order (the old code was backwards)